daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/DFDLSchemaFile.scala

```diff
@@ -137,9 +145,29 @@ final class DFDLSchemaFile(
 
   override lazy val optXMLSchemaDocument = iiParent.optXMLSchemaDocument
 
-  override lazy val uriString = schemaSource.uriForLoading.toString
+  override lazy val uriString = {
+    val uriString_ = schemaSource.uriForLoading.toString
+    val finalUriString = Misc.depersonalizePath(
+      uriString_
+        .replace(
+          Misc.classPath.find(cp => uriString_.startsWith(cp.toString)).getOrElse("").toString,
+          "",
+        ),
+    )
+    finalUriString
+  }
 
-  override protected lazy val diagnosticDebugNameImpl = schemaSource.uriForLoading.toString
+  override protected lazy val diagnosticDebugNameImpl = {
+    val uriString_ = schemaSource.uriForLoading.toString
+    val finalUriString = Misc.depersonalizePath(
+      uriString_
+        .replace(
+          Misc.classPath.find(cp => uriString_.startsWith(cp.toString)).getOrElse("").toString,
```

Comment:
This code is confusing. 

Can you spread this code out more? Add an intermediate for the uriString where the prefix that is found on the classpath has been removed, then depersonalize that in a separate line? The cascade of the replace with the find with the getOrElse is not easily understood.